cpp/src/arrow/acero/swiss_join.cc

```diff
@@ -643,37 +643,36 @@ void SwissTableMerge::MergePartition(SwissTable* target, const SwissTable* sourc
   //
   int source_group_id_bits =
       SwissTable::num_groupid_bits_from_log_blocks(source->log_blocks());
-  uint64_t source_group_id_mask = ~0ULL >> (64 - source_group_id_bits);
```

Comment:
After cleaning up these unnecessary 64-bit in this file, we can further cleanup some temp states as mentioned in https://github.com/apache/arrow/pull/45336#discussion_r1926710909 .